zipline/utils/sentinel.py

```diff
@@ -7,19 +7,33 @@
 
 
 def sentinel(name, doc=None):
-    @object.__new__   # bind a single instance to the name 'NotSpecified'
-    class result(object):
+    try:
+        value = sentinel._cache[name]  # memoized
+    except KeyError:
+        pass
+    else:
+        if doc == value.__doc__:
+            return value
+
+        raise ValueError(
+            'attempted to create sentinel with a used name and a different'
+            ' docstring: %r\noriginal docstring:\n%s' % (name, value.__doc__),
+        )
+
+    @object.__new__   # bind a single instance to the name 'Sentinel'
+    class Sentinel(object):
```

Comment:
Is there a reason for preferring this construction over a call to `type`?  It seems odd to create a class named `Sentinel` and then reset its name via __name__.